ddtrace/contrib/asgi/middleware.py

```diff
@@ -70,7 +70,11 @@ def _default_handle_exception_span(exc, span):
 def span_from_scope(scope):
     # type: (Mapping[str, Any]) -> Optional[Span]
     """Retrieve the top-level ASGI span from the scope."""
-    return scope.get("datadog", {}).get("request_span")
+    if "request_spans" in scope.get("datadog", {}):
```

Comment:
If this is the solution we've decided on then I think it's big enough to warrant a separate PR.

In that PR we should include the motivation of this change and test cases (independent of fastapi/starlette)

Reply:
https://github.com/DataDog/dd-trace-py/pull/3782